pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -771,6 +804,11 @@ func (c *Controller) AppendInstanceHandler(f func(*model.ServiceInstance, model.
 
 		c.updateEDS(ep, event)
 
+		instances := c.getServiceInstanceByEndpoint(*ep)
+		for _, instance := range instances {
+			f(instance, event)
```

Comment:
I realize the code is horrible, but I think this is going to trigger full pushes on endpoint changes now

@rshriram

Reply:
Oh yeah.. if you call the appendInstanceHandlers, it will trigger full push. Costin removed it but did not put in an alternative :(. Are you looking to handle the case where a new inbound listener appears? or the case where the inbound listener changes?

Reply:
I'm looking to handle the case of a new inbound listener. The instance handler I've added tries to trigger a push only for the proxy instance co-located with the new endpoint. Are you saying that the other instance handler, the one that simply calls `DiscoveryServer.clearCache()`, will trigger a full push to all proxies? If so, should it just be removed?

Reply:
I think we could just remove it. That is the only usage right?

Reply:
Yap, that's the only usage. Will remove.